### PR TITLE
Consolidate dataplane token code

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Use the same function for all dataplane token requests. This sets us up for storing and refreshing the dataplane token in a later PR.

### CODE REVIEW

File: internal/pkg/cmd/prerunner.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func (c *AuthenticatedCLICommand) GetFlinkGatewayClient() (*ccloudv2.FlinkGatewa
 			return nil, err
 		}
 
-		authToken, err := pauth.GetJwtTokenForV2Client(ctx.GetState(), ctx.GetPlatformServer())
+		authToken, err := pauth.GetDataplaneToken(ctx.GetState(), ctx.GetPlatformServer(), map[string]any{})
```
Comment (sgagniere) [inline internal/pkg/cmd/prerunner.go:152]: Could we replace `map[string[]any{}` with `nil`?

Comment (brianstrauch) [reply internal/pkg/cmd/prerunner.go:152]: If it gets encoded to `{}` in JSON, then sure. Let me investigate!

Comment (brianstrauch) [reply internal/pkg/cmd/prerunner.go:152]: Just checked and it doesn't work, we need to use an empty map

Comment (sgagniere) [reply internal/pkg/cmd/prerunner.go]: Well, that's unfortunate.

**Review by sgagniere — APPROVED** [review]

**Pull request merged**